llm/llama.go

```diff
@@ -464,6 +464,11 @@ func (llm *llama) Predict(ctx context.Context, prevContext []int, prompt string,
 		return err
 	}
 
+	// Remove first leading space from prevConvo if present
+	if len(prevConvo) > 0 && prevConvo[0] == ' ' {
+		prevConvo = prevConvo[1:]
+	}
```

Comment:
Thank you for reviewing the code. I chose manual string slicing for removing the initial space as it offers a performance advantage for this specific scenario. The operation is safe because we first check the length of the string before proceeding with the slice. If you think strings.TrimLeft would be a better fit, feel free to request a change.

Reply:
`strings.TrimPrefix` would be the equivalent since `strings.TrimLeft` will remove repeated spaces instead of just the first occurrence. That would be preferable since it's less error prone

Performance here isn't super important since it'll (likely) be offset by the less performant inference